### PR TITLE
Fix JsBuiltIn initialization code to not read root property

### PR DESCRIPTION
**Pull request by meg-gupta**
Also add assert so that in future we dont write js builtin code to access  any root property

Fixes OS#15467762

### CODE REVIEW

File: lib/Common/Exceptions/Throw.cpp [hunk 1]
```diff
@@ -82,6 +82,14 @@ namespace Js {
         RaiseException((DWORD)DBG_TERMINATE_PROCESS, EXCEPTION_NONCONTINUABLE, 0, NULL);
     }
 
+#ifdef ENABLE_JS_BUILTINS
+    void Throw::FatalJsBuiltInError()
+    {
+        AssertMsg(false, "Could not initialize JsBuiltIns!");
+        ReportFatalException(NULL, E_UNEXPECTED, Fatal_JsReentrancy_Error, 0);
```
Comment (MSLaguana) [inline lib/Common/Exceptions/Throw.cpp]: Should this have its own error reason instead of just `Fatal_JsReentrancy_Error`?

Comment (meg-gupta) [reply lib/Common/Exceptions/Throw.cpp]: Yes! Fixed

**Comment by meg-gupta** [comment]
Thanks for the review!

File: lib/Runtime/Base/FunctionBody.cpp [hunk 2]
```diff
@@ -76,8 +76,12 @@ namespace Js
         m_functionNumber(functionNumber),
         m_defaultEntryPointInfo(nullptr),
         m_displayNameIsRecyclerAllocated(false),
-        m_tag11(true)
+        m_tag11(true),
+        m_isJsBuiltInCode(false)
     {
+#if DBG
+        m_isJsBuiltInInitCode = false;
+#endif
```
Comment (jackhorton) [inline lib/Runtime/Base/FunctionBody.cpp]: nit: this (and the similar instances below) could still be done in the initializer list?

**Comment by meg-gupta** [comment]
The root property maybe overridden and when we inject the jsbuiltin init code, it may be use the overridden property. Same applies for Intl. I checked Intl, it doesn't load any root property other than the Intl object itself.

**Comment by meg-gupta** [comment]
@dotnet-bot test Ubuntu shared_ubuntu_linux_debug

**Pull request merged**